scripts/calver-init

```diff
@@ -1,19 +1,30 @@
 #!/usr/bin/env bash
+# Initialise calendar versioning scheme (pre-commit hook, merge driver, install.sh script)
+# Run on a repo that has at least one commit.
 
 set -e
+DU_MSG="### Text automatically added by daps-utils calver-init ###"
 
-function AddCalVerHook(){
-    mkdir .githooks &> /dev/null || echo ".githooks already exists"
-    touch .githooks/pre-commit
-    # Don't append text if it already exists
-    DU_MSG="### Text automatically added by daps-utils calver-init ###"
-    if grep -Fxq "${DU_MSG}" .githooks/pre-commit &> /dev/null
+function _AssertFirstInit(){
+    touch "$1"
+    if grep -Fxq "${DU_MSG}" "$1" &> /dev/null
     then
-	echo -e "ERROR: 'calver-init' has already been called. Manually remove previous edit by 'calver-init' from '.githooks/pre-commit' (i.e. the text between the long comment blocks), and then try again.\n" >&2
+	echo -e "ERROR: 'calver-init' has already been called. Manually remove previous edit by 'calver-init' from ""$1"" (i.e. the text between the long comment blocks), and then try again.\n" >&2
 	exit 1
     fi
-    echo '################################################################' >> .githooks/pre-commit
-    echo ${DU_MSG} >> .githooks/pre-commit
+}
+
+function _AddHeader(){
+    echo '################################################################' >> "$1"
+    echo "${DU_MSG}" >> "$1"
+    echo "$1" has been created or updated.
+}
+
+function AddCalVerHook(){
+    mkdir -p .githooks
+    # Don't append text if it already exists
+    _AssertFirstInit ".githooks/pre-commit"
+    _AddHeader ".githooks/pre-commit"
```

Comment:
When testing, I found that the following must have been missing (not sure why it even works without it...) to ensure that the correct shell is used. In my case, without it the precommit hook was using the wrong `echo` leading to `echo -n` printing the `-n` character (weird!) (as seen in [this issue](https://stackoverflow.com/questions/11193466/echo-n-prints-n))

```suggestion
    echo '#!/usr/bin/env bash' >> .githooks/pre-commit
    _AddHeader ".githooks/pre-commit"
```